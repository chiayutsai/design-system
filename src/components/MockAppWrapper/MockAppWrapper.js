import history from 'history'
import PropTypes from 'prop-types'
import App from 'components/App'
import configureStore from 'store/configureStore'
import { injectReducer } from 'store/makeRootReducer'

const initialState = {}

export const store = configureStore(initialState, { history })

const MockAppWrapper = ({ children }) => (
  <App
    context={{
      pathname: '',
      store,
      injectReducer,
    }}
    insertCss={() => {}}>
    {children}
  </App>
)

MockAppWrapper.propTypes = {
  children: PropTypes.node.isRequired,
}

export default MockAppWrapper
