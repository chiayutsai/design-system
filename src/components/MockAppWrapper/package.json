{
  "name": "MockAppWrapper",
  "version": "0.0.0",
  "private": true,
  "main": "./MockAppWrapper.js"
}
