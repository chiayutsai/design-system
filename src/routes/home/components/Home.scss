/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

@import '~components/variables';

.root {
  padding-left: 20px;
  padding-right: 20px;
}

.container {
  margin: 0 auto;
  padding: 0 0 40px;
  max-width: var(--max-content-width);

  .flex {
    display: flex;
    flex-wrap: wrap;
  }
}

.check {
  width: 100px;
  height: 100px;

  &.red {
    path {
      fill: red;
    }
  }
}

.check-div {
  width: 100px;
  height: 100px;
  background: url('./assets/check.svg') no-repeat center / contain;
}

.large-image {
  width: 460px;
  height: 300px;
  margin: 10px;

  &.background-image {
    background: url('./assets/artificial-intelligence-machine.jpg') no-repeat center / cover;
    flex-shrink: 0;
  }
}

.small-image {
  width: 64px;
  height: 64px;
  margin: 4px;

  &.background-image {
    background: url('./assets/positive-vote.png') no-repeat center / cover;
  }
}

.hoverable {
  &:hover {
    color: #0093FC;
  }
}
