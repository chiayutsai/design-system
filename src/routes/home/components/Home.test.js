import renderer from 'react-test-renderer'
import MockAppWrapper, { store } from 'components/MockAppWrapper'
import setupLocale from 'test/utils/setupLocale'
import Home from './Home'

describe('Home', () => {
  // setup i18next if component needed
  setupLocale(store)

  test('render correctly', () => {
    const wrapper = renderer
      .create(
        <MockAppWrapper>
          <Home />
        </MockAppWrapper>,
      )
      .toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
