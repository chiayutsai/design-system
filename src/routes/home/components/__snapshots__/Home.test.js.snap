// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home render correctly 1`] = `
<div
  className="root"
>
  <div
    className="container"
  >
    <h1>
      React.js News
    </h1>
    <h2>
      i18n test
    </h2>
    <p>
      Betting
    </p>
    <h2>
      svg test
    </h2>
    <div
      className="flex"
    >
      <div
        className="check"
      />
      <div
        className="check red"
      />
      <img
        alt="check"
        className="check"
        src="SvgrURL"
      />
      <div
        className="check-div"
      />
    </div>
    <h2>
      loader test
    </h2>
    <h3>
      large image
    </h3>
    <div
      className="flex"
    >
      <div
        className="large-image background-image"
      />
      <img
        alt=""
        className="large-image"
        src="artificial-intelligence-machine.jpg"
      />
    </div>
    <h3>
      small image
    </h3>
    <div
      className="flex"
    >
      <div
        className="small-image background-image"
      />
      <img
        alt=""
        className="small-image"
        src="positive-vote.png"
      />
    </div>
    <h3>
      json
    </h3>
    {"glossary":{"title":"example glossary","GlossDiv":{"title":"S","GlossList":{"GlossEntry":{"ID":"SGML","SortAs":"SGML","GlossTerm":"Standard Generalized Markup Language","Acronym":"SGML","Abbrev":"ISO 8879:1986","GlossDef":{"para":"A meta-markup language, used to create markup languages such as DocBook.","GlossSeeAlso":["GML","XML"]},"GlossSee":"markup"}}}}}
    <h3>
      txt
    </h3>
    
    <h2>
      postcss loader test:
    </h2>
    <h3
      className="hoverable"
    >
      hover me!
    </h3>
  </div>
</div>
`;
